apm-agent-core/src/main/java/co/elastic/apm/agent/report/ssl/SslUtils.java

```diff
@@ -108,19 +113,28 @@ public static SSLSocketFactory getSSLSocketFactory(boolean validateCertificates)
     }
 
     @Nullable
-    private static SSLSocketFactory createSocketFactory(TrustManager[] trustAllCerts) throws NoSuchAlgorithmException, KeyManagementException {
+    private static SSLSocketFactory createSocketFactory(@Nullable TrustManager[] trustManagers) throws IOException, GeneralSecurityException {
         SSLContext sslContext;
         try {
             sslContext = SSLContext.getInstance("SSL");
         } catch (NoSuchAlgorithmException e) {
             logger.info("SSL is not supported, trying to use TLS instead.");
             sslContext = SSLContext.getInstance("TLS");
         }
-        sslContext.init(null, trustAllCerts, new java.security.SecureRandom());
+
+        KeyManager[] keyManagers;
+        try {
+            keyManagers = getKeyManagers();
+        } catch (IOException | GeneralSecurityException e) {
+            logger.error("unable to setup key managers", e);
+            throw e;
```

Comment:
do we want to error here, or just revert to previous behaviour with null keymanagers since most of our customers don't need this?

Reply:
That's a very good point, here issuing a warning and then silently use the previous behavior seems appropriate.